components/backup/tests/integrations/test_backup.rs

```diff
@@ -457,14 +489,102 @@ fn test_backup_and_import() {
         &make_unique_dir(tmp.path()),
     );
     let resps3 = block_on(rx.collect::<Vec<_>>());
-    assert_eq!(files1, resps3[0].files);
+    assert_same_files(files1.into_vec(), resps3[0].files.clone().into_vec());
+
+    suite.stop();
+}
+
+#[test]
+fn test_backup_huge_range_and_import() {
+    let mut suite = TestSuite::new(3, 100);
+    // 3 version for each key.
+    // make sure we will have two batch files
+    let key_count = 1024 * 3 / 2;
+    suite.must_kv_put(key_count, 3);
+
+    // Push down backup request.
+    let tmp = Builder::new().tempdir().unwrap();
+    let backup_ts = suite.alloc_ts();
+    let storage_path = make_unique_dir(tmp.path());
+    let rx = suite.backup(
+        vec![],   // start
+        vec![],   // end
+        0.into(), // begin_ts
+        backup_ts,
+        &storage_path,
+    );
+    let resps1 = block_on(rx.collect::<Vec<_>>());
+    // Only leader can handle backup.
+    assert_eq!(resps1.len(), 1);
+    let files1 = resps1[0].files.clone();
+    // Short value is piggybacked in write cf, so we get 1 sst at least.
+    assert!(!resps1[0].get_files().is_empty());
+    assert_eq!(files1.len(), 2);
```

Comment:
Please make sure these two files have different start/end key.

Reply:
addressed in https://github.com/tikv/tikv/pull/9283/commits/0601fbb192d932a4469ba2bafa075c7883418292